pkg/addons/addons.go

```diff
@@ -200,6 +208,18 @@ func writeDB(client Interface, db map[string]unstructured.Unstructured) error {
 	if err := client.ApplyResources(cfgFilter, db, keys); err != nil {
 		return err
 	}
+
+	log.Debug("Waiting for the Etcd CRD to be ready")
+	if err := wait.PollInfinite(time.Second, client.EtcdCRDReady); err != nil {
+		return err
+	}
+	log.Debug("Etcd CRD is ready")
+
+	// refresh dynamic client
+	if err := client.UpdateDynamicClient(); err != nil {
```

Comment:
If we wrap `EtcdCRDReady` and `ServiceCatalogExists` in subfunctions (or rewrite them) we can move update client logic into the clients itself. Methods should know if they need to update client or not. follow-up issue?

Reply:
follow up.  Taggiong

Reply:
https://github.com/openshift/openshift-azure/issues/380